From a50f7ea2dd023ef124c5209b487608b37dbeea7c Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sat, 1 Mar 2025 19:33:18 +0100
Subject: [PATCH] io_url_wget: correctly init wget_out on apk_io_url_init call

Commit fc7768c09497 ("io_url_wget: log wget output using apk_process
api") reworked the implementation but forgot to init the static wget_out
on calling apk_io_url_init. This cause APK to SIGFAULT on apk update or
apk add when actually using the WGET method to download remote packages.

Fix this by setting wget_out with the passed out from apk_io_url_init.

Reported-by: John Crispin <john@phrozen.org>
Fixes: fc7768c09497 ("io_url_wget: log wget output using apk_process api")
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 src/io_url_wget.c | 1 +
 1 file changed, 1 insertion(+)

--- a/src/io_url_wget.c
+++ b/src/io_url_wget.c
@@ -48,4 +48,5 @@ void apk_io_url_set_redirect_callback(vo
 
 void apk_io_url_init(struct apk_out *out)
 {
+	wget_out = out;
 }
